ee_plugin/ui/forms/add_ee_image.py

```diff
@@ -0,0 +1,101 @@
+import json
+from typing import Callable, Optional
+
+import ee
+from qgis.PyQt import QtWidgets
+from qgis.core import QgsMessageLog, Qgis
+
+from ...Map import addLayer
+from ..widgets import build_form_group_box, build_vbox_dialog
+from ..utils import call_func_with_values
+
+
+def form(accepted: Optional[Callable] = None, **dialog_kwargs) -> QtWidgets.QDialog:
+    """Display a dialog to add a GEE dataset to the QGIS map."""
+    dialog = build_vbox_dialog(
+        windowTitle="Add Google Earth Engine Image",
+        widgets=[
+            build_form_group_box(
+                title="Dataset",
+                rows=[
+                    (
+                        QtWidgets.QLabel(
+                            text="Enter GEE Image Name (e.g., COPERNICUS/S2, USGS/SRTMGL1_003)",
+                            toolTip="Provide the full Earth Engine ID.",
+                        ),
+                        QtWidgets.QLineEdit(objectName="image_id"),
+                    )
+                ],
+            ),
+            build_form_group_box(
+                title="Visualization Parameters (JSON)",
+                collapsable=True,
+                collapsed=True,
+                rows=[
+                    (
+                        QtWidgets.QLabel(
+                            text="Enter JSON for visualization parameters",
+                            toolTip="Example: {'min': 0, 'max': 4000, 'palette': ['006633', 'E5FFCC', '662A00']}",
+                        ),
+                        QtWidgets.QTextEdit(
+                            objectName="viz_params",
+                            placeholderText='{"min": 0, "max": 4000, "palette": ["006633", "E5FFCC", "662A00"]}',
+                        ),
+                    )
+                ],
+            ),
+        ],
+        **dialog_kwargs,
```

Comment:
I added this to allow further customization from the parent if desired.  Not sure if that just confuses things...